draft-ietf-sidrops-aspa-verification.xml

```diff
@@ -737,36 +824,100 @@ hop(AS(i), AS(j)) =  / Else, "Provider+" if the VAP-SPAS entry
+      AS(1) is originating the prefix p.
+      Normally, the receiving/verifying AS (AS(5)) should receive the route for prefix p with AS_PATH {AS(4), AS(3), AS(2), AS(1)} and it would be Valid (<xref target="Downflow"/>) given all the ASPAs that are shown.
+      But AS(4) which is a transit provider of AS(5) manipulates the AS_PATH as follows.
+      There is no physical connection or relationship between AS(4) and AS(2), but AS(4) fakes that connection and sends a route with a shortened AS_PATH {AS(4), AS(2), AS(1)}.
+      Based on the ASPA-based path verification, AS(5) cannot detect that this AS_PATH is Invalid.
+      The verification algorithm at AS(5) inherently infers that AS(4) is a customer or lateral peer of AS(2) (i.e., not a provider) because AS(4) is not included in AS(2)'s ASPA.
+      Also, if AS(4) had performed a forged-origin hijack by inserting an AS_PATH {AS(4), AS(1)}, that would also be undetectable.
+      A transit provider is in a position to manipulate AS_PATHs towards its customer in this manner and avoid detection.
+      The motivation may be to announce a shorter path to attract customer traffic away from an alternate provider.
+      In <xref target="fig3"/>, AS(5)'s alternate path to prefix p is {AS(6), AS(3), AS(2), AS(1)} which is also Valid but longer than the path received from AS(4).
+    </t>
+    <t>
+      <figure anchor="fig3" align="left" suppress-title="false" pn="figure-4">
+        <name slugifiedName="attack1">Illustration for discussion of undetectable AS_PATH manipulations.</name>
+        <artwork align="center" name="" type="" alt="">
+<![CDATA[
+
+                                 AS(3)
+          _____<--P2C___________/  /  \______<--C2P_______
+         /                        /                       \
+       AS(6)                     /p{AS(3), AS(2), AS(1)}   \
+         \                      /                           \
+          \                   AS(4)----faked connection----AS(2)
+           \__P2C-->__           /                             \
+                      \         / (down-ramp)         (up-ramp) \
+                       \       /                                 \
+     p{AS(6), AS(3),    \     /p{AS(4), AS(2), AS(1)}          AS(1)
+          AS(2), AS(1)}  \   /  (path manipulated)    (Origin AS) |
+                        AS(5)                                     p
+                  Receiving & verifying AS                  (prefix)
+
+ASPAs: {AS(1), [AS(2)]}, {AS(2), [AS(3)]}, {AS(3), [AS 0]},
+       {AS(4), AS(3)}, {AS(5), AS(4)}, {AS(6), [AS(5)]}
+]]>
+        </artwork>
+      </figure>
+    </t>
+    <t>
+      In the above scenario, it may also be noted that if the attacking AS forwards the data traffic destined
+      for the affected prefix, and if all other ASes physically connected to the attacking AS are ASPA compliant,
+      then the data traffic (from the deceived customer) would flow on a feasible route-leak free path in spite of the attack.
```

Comment:
I don't think there is a need to explain which path manipulations can and which can't be detected by ASPA.
The detection of path manipulations is more of a side-effect of what ASPA provides but I would not sell ASPA as secure.
It makes it harder which is great but it feels out of scope for this document.

There is a lot of text added about various kind of hijacks. I don't think ASPA really provides a major hurdle here -- it helps and raises the bar for sure but like ROA it does not prevent hijacks.
Especially sophisticated attacks done by cyber criminals and hacker groups.

Reply:
I was indeed in doubt about including the specific paragraph that you highlighted. Will remove it. 

>There is a lot of text added about various kind of hijacks. I don't think ASPA really provides a major hurdle here -- it helps and raises the bar for sure but like ROA it does not prevent hijacks.

What you are stating is not entirely correct. We have known it all along and WG members certainly appreciated that ASPA does prevent forged-origin and forged-path-segment hijacks in the *upstream paths*. But the same is not true for the downstream paths. Folks understood and liked the *upstream paths* part and they appreciated the text about that during WGLC discussions.  

WGLC participants also liked highlighting an example of undetected AS path manipulation in the Security Considerations section. Rudiger strongly advised including multiple examples because he thought only one example was inadequate and misleading. There are in indeed at least three distinct types that are undetected by ASPA, and I have included them in the Security Considerations section.  

Thank you, Claudio, for the comments. Very helpful

Reply:
I do understand that ASPA prevents some form of forged hijacks but in my opinion it gives a false sense of security.

It only works in some cases (_upstream paths_) and only if all possible providers reached in the graph from the origin-AS up to tier-1 AS use ASPA. Right now a hijack can be done via unknown hops (provider with no ASPA record published) which is sadly the most common case right now and for the foreseeable future.

It is a great side-effect and one of the reasons I really push for ASPA to become widespread but I think the bar for the Security Considerations section should be higher.